src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLoggerExtensions.cs

```diff
@@ -2,16 +2,131 @@
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
 using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Logging;
 using Microsoft.Azure.WebJobs.Logging.ApplicationInsights;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Extensions.Logging
 {
     /// <summary>
-    /// Extensions for adding the <see cref="ApplicationInsightsLoggerProvider"/> to an <see cref="ILoggerFactory"/>. 
+    /// Extensions for ApplicationInsights configurationon an <see cref="IHostBuilder"/>. 
     /// </summary>
     public static class ApplicationInsightsLoggerExtensions
     {
+        /// <summary>
+        /// Registers Application Insights and <see cref="ApplicationInsightsLoggerProvider"/> with an <see cref="IHostBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The host builder.</param>
+        /// <param name="instrumentationKey">The Application Insights instrumentation key.</param>
+        /// <returns>A <see cref="IHostBuilder"/> for chaining additional operations.</returns>
+        public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, string instrumentationKey)
+        {
+            return AddApplicationInsights(builder, instrumentationKey, (_, level) => level > LogLevel.Debug);
+        }
+
+        /// <summary>
+        /// Registers Application Insights and <see cref="ApplicationInsightsLoggerProvider"/> with an <see cref="IHostBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The host builder.</param>
+        /// <param name="instrumentationKey">The Application Insights instrumentation key.</param>
+        /// <param name="filter">A filter that returns true if a message with the specified <see cref="LogLevel"/>
+        /// and category should be logged. You can use <see cref="LogCategoryFilter.Filter(string, LogLevel)"/>
+        /// or write a custom filter.</param>
+        /// <returns>A <see cref="IHostBuilder"/> for chaining additional operations.</returns>
+        public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, string instrumentationKey, Func<string, LogLevel, bool> filter)
+        {
+            if (string.IsNullOrEmpty(instrumentationKey))
+            {
+                return builder;
+            }
+
+            builder.ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<ITelemetryInitializer, WebJobsRoleEnvironmentTelemetryInitializer>();
+                services.AddSingleton<ITelemetryInitializer, WebJobsTelemetryInitializer>();
+                services.AddSingleton<ITelemetryInitializer, WebJobsSanitizingInitializer>();
```

Comment:
adding multiple services with the same service type is ok (while they have different implementation type).

It's also possible to use `TryAddEnumerable` (thanks, I did not know about it!)

```csharp
services.TryAddEnumerable(new[] {
    ServiceDescriptor.Describe(serviceType: typeof(ITelemetryInitializer), implementationType: typeof(WebJobsRoleEnvironmentTelemetryInitializer), lifetime: ServiceLifetime.Singleton),
    ServiceDescriptor.Describe(serviceType: typeof(ITelemetryInitializer), implementationType: typeof(WebJobsTelemetryInitializer), lifetime: ServiceLifetime.Singleton),
    ServiceDescriptor.Describe(serviceType: typeof(ITelemetryInitializer), implementationType: typeof(WebJobsSanitizingInitializer), lifetime: ServiceLifetime.Singleton)
});
```

But i think it looks more complicated and verbose, so I'd prefer to keep current implementation

Reply:
If `AddSingleton` works, by all means use it. We didn't think that this call would work: `provider.GetServices<ITelemetryInitializer>()` unless you tried to register it as an enumerable. I thought that it would overwrite your previous service registration and leave you with only a single `ITelemetryInitializer`. But it looks like it does work. Nice :-)